### ISSUE TITLE
confusing about stop_gradient in the code

### ISSUE THREAD

**Issue by kelvinleen** [google-research/electra#22]
why there are two stop_gradients in "mask" function of pretrain_helpers.py. see snapshot under. 
![image](https://user-images.githubusercontent.com/14268422/77148763-23d09900-6acb-11ea-9b19-46d30fae32b5.png)

i can understand the stop_gradient here, but after that, there is still an argmax op, seems strange
![image](https://user-images.githubusercontent.com/14268422/77148816-419dfe00-6acb-11ea-964c-79569911bb13.png)

**Comment by clarkkev** [comment]
The `tf.stop_gradients` were there to stop back-propagating through non-differentiable operations (categorical sampling and argmax), although I'm not 100% sure if they are actually necessary, as tensorflow may not backprop through them by default.
